packages/itwin/grouping-mapping-widget/src/widget/components/MappingsUI.tsx

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import {
+  Button,
+  IconButton,
+} from "@itwin/itwinui-react";
+import {
+  SvgAdd,
+  SvgImport,
+  SvgRefresh,
+} from "@itwin/itwinui-icons-react";
+import {
+  EmptyMessage,
+  LoadingOverlay,
+} from "./utils";
+import "./MappingsUI.scss";
+import DeleteModal from "./DeleteModal";
+import { MappingImportWizardModal } from "./MappingImportWizardModal";
+import { HorizontalTile } from "./HorizontalTile";
+import type { Mapping } from "@itwin/insights-client";
+import { BlockingOverlay } from "./BlockingOverlay";
+import { MappingUIActionGroup } from "./MappingUIActionGroup";
+
+export const mappingUIDefaultDisplayStrings = {
+  mappings: "Mappings",
+  iTwins: "iTwins",
+  iTwinNumber: "Number",
+  iTwinName: "Name",
+  iTwinStatus: "Status",
+  iModels: "iModels",
+  iModelName: "Name",
+  iModelDescription: "Description",
+};
+
+export interface MappingsUIProps {
+  mappings: Mapping[];
+  isLoading: boolean;
+  isTogglingExtraction: boolean;
+  onRefresh: () => Promise<void>;
+  onToggleExtraction: (mapping: Mapping) => Promise<void>;
+  onDelete: (mapping: Mapping) => Promise<void>;
+  showDeleteModal: Mapping | undefined;
+  setShowDeleteModal: (mapping?: Mapping) => void;
+  displayStrings?: Partial<typeof mappingUIDefaultDisplayStrings>;
+  showImportModal?: boolean;
+  setShowImportModal?: (show: boolean) => void;
+  onClickAddMapping?: () => void;
+  onClickMappingTitle?: (mapping: Mapping) => void;
+  onClickMappingModify?: (mapping: Mapping) => void;
+}
+
+export const MappingsUI = ({
+  mappings,
+  isLoading,
+  isTogglingExtraction,
+  onRefresh,
+  onToggleExtraction,
+  onDelete,
+  showDeleteModal,
+  setShowDeleteModal,
+  displayStrings: userDisplayStrings,
+  showImportModal,
+  setShowImportModal,
+  onClickAddMapping,
+  onClickMappingTitle,
+  onClickMappingModify,
+}: MappingsUIProps) => {
+  const displayStrings = React.useMemo(
+    () => ({ ...mappingUIDefaultDisplayStrings, ...userDisplayStrings }),
+    [userDisplayStrings]
+  );
+
+  return (
+    <>
+      <BlockingOverlay isVisible={isTogglingExtraction} />
+      <div className="gmw-mappings-container">
```

Comment:
Just a suggestion, I find the itwinui <Flex> component useful, especially with nested divs. Besides reducing CSS, I find the JSX becomes more "readable".

Reply:
Yup, the iTwinUI flex component is very nice. I was thinking of eventually replacing the CSS later on with that.